### ISSUE TITLE
Simplify SqlServerDatabase Resource Type

### ISSUE THREAD

**Issue by afscrome** [dotnet/aspire#5555]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

`Project`, `Container` and `Executable` resoruces have nice concice names in the `Type` column of the Resources page (and in the `Resource Types` filter).
<img width="897" alt="image" src="https://github.com/user-attachments/assets/f55e0776-7b3a-40c8-9b6c-1295f7eeed02">

### Describe the solution you'd like

It would be nice if `SqlServerDatabaseResource` (and possibly other resources) could have a similarly concise name.  (e.g. `SqlDatabase` or `SqlServerDatabase`)

**Comment by maddymontaquila** [comment]
Gonna close this as a dupe of #2595  since we'd ideally address this as part of that work. Great call though - completely agree 💯

**Comment by maddymontaquila** [comment]
Another option is to just make custom resources look nicer with a lil change so we could do that in the meantime

**Comment by JamesNK** [comment]
@mitchdenny @davidfowl Where does this name come from? Is it the type name? If so, could a fix be a simple as removing "Resource" suffix from the type name?

**Comment by davidfowl** [comment]
We can do that, but it seemed a little aggressive to assume 😄:

https://github.com/dotnet/aspire/blob/47e3e25e51c60627aa5fa9a4bf4aaeaf26402ce2/src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs#L348

**Comment by JamesNK** [comment]
What is the alternative? Add a ResourceTypeNameAnnotation?

**Comment by davidfowl** [comment]
Resources can already set it using `WithInitialState`

**Comment by eerhardt** [comment]
What about a declarative model where we add a normal .NET attribute to the class? If the attribute is present, use it. If it isn't, fallback to the TypeName.

That seems easier than littering all the code with calls to `WithInitialState` just to set the `string ResourceType` property.

Another thing to consider here is that if we call `WithInitialState` on all the resources, we are going to miss out on this Relationship building code:

https://github.com/dotnet/aspire/blob/fd1853aa19602df86a0a4ee7d077356187c6bbdf/src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs#L491-L497

If there is already a snapshot, that code won't be run.

**Comment by JamesNK** [comment]
I like using an attribute. Use .NET's existing `DescriptionAttribute`?

**Comment by eerhardt** [comment]
Good idea using an existing attribute. I think [DisplayNameAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.displaynameattribute) should work good for this.

**Comment by davidfowl** [comment]
DisplayName is better. I would prefer something that could be changed dynamically without requiring inheritance but that doesn't seem like a bad idea to allow this. Plus you can always explicitly set it 😄
